dsp/modules/unify.py

```diff
@@ -1,38 +1,32 @@
 import logging
-from typing import Any, Literal, Optional
+from typing import Any, Optional
 
 from unify.clients import Unify as UnifyClient
 
 from dsp.modules.lm import LM
 
 
-class Unify(LM):
+class Unify(LM, UnifyClient):
     """A class to interact with the Unify AI API."""
 
     def __init__(
         self,
-        endpoint="router@q:1|c:4.65e-03|t:2.08e-05|i:2.07e-03",
-        # model: Optional[str] = None,
-        # provider: Optional[str] = None,
-        model_type: Literal["chat", "text"] = "chat",
+        endpoint: str = "router@q:1|c:4.65e-03|t:2.08e-05|i:2.07e-03",
+        model: Optional[str] = None,
+        provider: Optional[str] = None,
+        api_key=None,
         stream: Optional[bool] = False,
-        base_url="https://api.unify.ai/v0",
         system_prompt: Optional[str] = None,
+        base_url: str = "https://api.unify.ai/v0",
         n: int = 1,
-        api_key=None,
         **kwargs,
     ):
-        self.api_key = api_key
-        # self.model = model
-        # self.provider = provider
-        self.endpoint = endpoint
+        self.base_url = base_url
         self.stream = stream
-        self.client = UnifyClient(api_key=self.api_key, endpoint=self.endpoint)
-
-        super().__init__(model=self.endpoint)
-
+        LM.__init__(self, model)
+        UnifyClient.__init__(self, endpoint=endpoint, model=model, provider=provider, api_key=api_key)
+        # super().__init__(model)
         self.system_prompt = system_prompt
-        self.model_type = model_type
```

Comment:
reason: no model_type option because generate in the Unify class is always using "chat.completions()". Other clients don't offer this option either.

Reply:
Yes, that's correct.